.github/workflows/emerynet-devnet-test.yml

```diff
@@ -0,0 +1,104 @@
+        run: docker-compose -f docker-compose-testnet.yaml up -d
+
+      - name: Check if Hermes relayer has started
+        run: |
+          timeout 420 bash -c "\
+          until docker logs relayer 2>&1 | grep -q 'Hermes has started'; do
+            echo 'waiting for relayer to start...'
+          done"
+
+      - name: Check Emerynet Wallet Balance
+        run: |
+          curl https://emerynet.api.agoric.net/cosmos/bank/v1beta1/balances/$WALLET_ADDRESS_EMERYNET | 
+          jq -r '.balances[] | select(.denom == "ubld") | .amount | tonumber' > $EMERYNET_BALANCE_FILE
+
+      - name: Verify Token Count in Local Chain Wallet
+        run: |
+          curl http://localhost:1317/cosmos/bank/v1beta1/balances/$WALLET_ADDRESS_LOCAL | 
+          jq -r '.balances | length' > $LOCAL_BALANCE_FILE
+
+      - name: Monitor Channel Initialization
+        run: |
+          timeout 360 bash -c "\
+          until line=\$(docker logs relayer 2>&1 | grep 'OpenInitChannel'); do
+            echo 'Waiting for channel initialization...'
+            sleep 1
+          done
+          if [[ -n \$line ]]; then
+            channel_id=\$(echo \$line | awk -F'channel_id: ' '{print \$2}' | awk -F', ' '{print \$1}')
+            echo \"Channel ID extracted: \$channel_id\"
+            echo \$channel_id > channel_id.txt
+          else
+            echo 'Channel initialization timed out.'
+            exit 1
+          fi"
+
+      - name: Confirm Channel Status
+        run: |
+          channel_id=$(cat channel_id.txt)
+          timeout 600 bash -c "\
+          until docker logs relayer 2>&1 | grep -q 'OpenConfirmChannel.*$channel_id'; do
+            echo 'Awaiting confirmation for channel ID: $channel_id...'
+            sleep 1
+          done
+          echo 'Channel ID $channel_id confirmed.'"
+
+      - name: Execute IBC Token Transfer
+        run: |
+          docker exec relayer hermes --config $RELAYER_CONFIG tx ft-transfer --src-chain $EMERYNET_CHAIN_ID --src-channel $(cat channel_id.txt) \
+          --dst-chain $LOCAL_CHAIN_ID --src-port transfer --amount 100 --denom 'ubld' --timeout-seconds 1000 && \
+          sleep 5
+
+      - name: Verify Token Transfer on Emerynet
+        run: |
+          NEW_EMERYNET_BALANCE=$(curl https://emerynet.api.agoric.net/cosmos/bank/v1beta1/balances/$WALLET_ADDRESS_EMERYNET | 
+          jq -r '.balances[] | select(.denom == "ubld") | .amount | tonumber')
+          ORIGINAL_EMERYNET_BALANCE=$(cat $EMERYNET_BALANCE_FILE)
+          BALANCE_DIFFERENCE=$((ORIGINAL_EMERYNET_BALANCE - NEW_EMERYNET_BALANCE))
+          if [ "$BALANCE_DIFFERENCE" -ne 100 ]; then
+            echo "Token transfer discrepancy detected: Expected 100, found $BALANCE_DIFFERENCE."
+            exit 1
+          else
+            echo "Token transfer validated: Balance difference is exactly 100ubld."
+          fi
+
+      - name: Verify Token Transfer on Local Chain
+        run: |
+          sleep 120
```

Comment:
Is there a way to poll for status updates on the transaction instead of sleeping?

Reply:
Yes, made [this update](https://github.com/agoric-labs/agoric-ibc-relayer/pull/1/files#:~:text=fi-,%2D%20name%3A%20Verify%20Token%20Transfer%20on%20Local%20Chain,%27,-20%20changes%3A%2019)